src/mympi.cpp

```diff
@@ -147,10 +147,13 @@ void send(int from, int to, double *data, int size) {
 }
 
 #if MEEP_SINGLE
-void broadcast(int from, realnum *data, int size) {
+void broadcast(int from, realnum *data, meep::integer size) {
 #ifdef HAVE_MPI
   if (size == 0) return;
-  MPI_Bcast(data, size, MPI_FLOAT, from, mycomm);
+  if (size > INT32_MAX){
+ 	  abort("broadcast size exeeds maximum message size");
```

Comment:
Since you have this check several times, it would be better to write a function:
```c
static void check_size(meep::integer size) {
  if (size > INT32_MAX) abort("MPI message size exceeds maximum");
}
```